ledger/src/blockstore.rs

```diff
@@ -5390,6 +5401,62 @@ impl Blockstore {
         Ok(())
     }
 
+    /// Handles chaining updates when an UpdateParent marker overrides a previously
+    /// set parent from a BlockHeader.
+    ///
+    /// TODO(ashwin, ksn): this chaining only occurs for `SlotMeta`s in the column associated with
+    /// `BlockLocation::Original`
+    fn update_chaining_for_parent_metas(
+        &self,
+        working_set: &mut HashMap<(BlockLocation, u64), SlotMetaWorkingSetEntry>,
+        parent_metas: &HashMap<(BlockLocation, u64), ParentMetaWorkingSetEntry>,
+        new_chained_slots: &mut HashMap<u64, Rc<RefCell<SlotMeta>>>,
+    ) -> Result<()> {
+        // Filter to dirty UpdateParent entries for Original location only
+        let update_parent_entries = parent_metas.iter().filter_map(|((location, slot), wse)| {
+            if !matches!(location, BlockLocation::Original) {
+                return None;
+            }
+
+            match &wse.parent_meta {
+                Some(WorkingEntry::Dirty(pm)) if pm.populated_from_update_parent() => {
+                    Some((*slot, pm.parent_slot))
+                }
+                _ => None,
+            }
+        });
+
+        for (slot, new_parent_slot) in update_parent_entries {
+            // Get old parent from slot meta; update parent_slot to new value
+            let old_parent_slot =
+                if let Some(entry) = working_set.get_mut(&(BlockLocation::Original, slot)) {
+                    let old = entry.old_slot_meta.as_ref().and_then(|m| m.parent_slot);
```

Comment:
do we have a test where there is an UpdateParent and a BlockHeader parent in the same shred insertion batch, testing both orders where

1. BlockHeader precedes UpdateParent
2. UpdateParent precedes BlockHeader